assets/src/application/ui/forms/espressoForm/renderers/FieldRenderer.tsx

```diff
@@ -39,9 +39,7 @@ const FieldRenderer: React.FC<FieldRendererProps> = (props) => {
 			<FormLabel htmlFor={props.input.name}>
 				{label}
 				{info && (
-					<Tooltip placement='right' tooltip={info}>
-						<InfoCircleOutlined />
-					</Tooltip>
+					<ClickableIconWithTooltip icon={HelpIcon} tooltipText={info} />
```

Comment:
I'm thinking it might be better to convert `ClickableIconWithTooltip` into an HOC and then wrap the export of `HelpIcon` with it.

Then this client code would only need to import and use `HelpIcon` and not have to deal with configuration issues pertaining to `ClickableIconWithTooltip`.

IMHO that will make things more readable and will also make it easier to make changes to the tooltip aspect of `HelpIcon` since it will only require updates to the one file.

But I'm open to debate on this in case there is a good reason to keep things as is.

thoughts?

Reply:
Just today @manzoorwanijk pointed out that one of the HOC doesn't have typed return and applying that kind of types is tricky/ complicated. Then I noticed that all of ours HOCs doesn't have return type and created an issue to solve this problem later: https://github.com/eventespresso/barista/projects/1#card-40726511

Reply:
really?
I'm seeing this:

```
const withLabel: <P extends withLabelProps>(WrappedComponent: React.ComponentType<P>) => React.ForwardRefExoticComponent<React.PropsWithoutRef<P> & React.RefAttributes<React.ComponentType<P>>>
```

which looks to me like it's returning `React.ForwardRefExoticComponent`

Reply:
basically the return types are being inferred from:

```js
return React.forwardRef(ForwardedComponentWithTooltip);
```

so the return types are not hard coded, but they are still there nonetheless

Reply:
I think @manzoorwanijk can have something to say here since he originally raised the issue

Reply:
Some of our HOCs (not all) do not have the return type defined. Though TS infers it most of the time but we should be explicit wherever possible.

Reply:
If a HOC uses generics and then uses those generics in its underlying types, then it's difficult to set the return type. In such cases, we can leave it to TS to infer.

Reply:
is `React.ForwardRefExoticComponent` sufficient though or was your original comment because we should be more explicit so that the return types show the specific component being used (ie: via generics)?